syntax = "proto3";

package models;

import "google/protobuf/timestamp.proto";

message User {
  int32 id = 1;
  string name = 2;
  string mail = 3;
  bool blocked = 5;
  google.protobuf.Timestamp created_at = 6;
  google.protobuf.Timestamp updated_at = 7;
  int32 role_id = 8;
  google.protobuf.Timestamp last_login = 9;
  string last_login_ip = 10;
}

service UserService {
  rpc CreateUser (CreateUserRequest) returns (User);
  rpc GetUser (GetUserRequest) returns (User);
  rpc GetUsersByRoleId (GetUsersByRoleIdRequest) returns (UsersResponse);
  rpc UpdateUserLastLogin (UpdateUserLastLoginRequest) returns (User); // Added specific service method
}

message CreateUserRequest {
  int32 id = 1;
  string name = 2;
  string mail = 3;
  bool blocked = 5;
  int32 role_id = 6;
  string last_login_ip = 7;
}

message GetUserRequest {
  int32 id = 1;
}

message GetUsersByRoleIdRequest {
  int32 role_id = 1;
}

message UpdateUserLastLoginRequest {
  int32 id = 1;
  google.protobuf.Timestamp last_login = 2;
}

message UsersResponse {
  repeated User users = 1;
}
